pp_netlib/gt_prune.py

```diff
@@ -0,0 +1,293 @@
+import sys
+import operator
+import numpy as np
+from scipy.stats import rankdata
+from pp_netlib.utils import gen_unword, read_isolate_type_from_csv, print_external_clusters
+import graph_tool.all as gt
+from collections import defaultdict, Counter
+
+def get_gt_clique_refs(graph, reference_indices:set()):
+    """Recursively prune a network of its cliques. Returns one vertex from
+    a clique at each stage
+
+    ## DEPENDS ON Fns: {none}
+
+    Args:
+        graph (graph)
+            The graph to get clique representatives from
+        reference_indices (set)
+            The unique list of vertices being kept, to add to
+    """
+    cliques = gt.max_cliques(graph)
+    try:
+        # Get the first clique, and see if it has any members already
+        # contained in the vertex list
+        clique = frozenset(next(cliques))
+        if clique.isdisjoint(reference_indices):
+            reference_indices.add(list(clique)[0])
+
+        # Remove the clique, and prune the resulting subgraph (recursively)
+        subgraph = gt.GraphView(graph, vfilt=[v not in clique for v in graph.vertices()])
+        if subgraph.num_vertices() > 1:
+            get_gt_clique_refs(subgraph, reference_indices)
+        elif subgraph.num_vertices() == 1:
+            reference_indices.add(subgraph.get_vertices()[0])
+    except StopIteration:
+        pass
+
+    return reference_indices
+
+def gt_clique_prune(component, graph, reference_indices, components_list):
```

Comment:
Multiprocessing should be used to wrap the call to this across components I think?